### PR TITLE
re-enabled decision tree classifier

### PR DESCRIPTION
**Pull request by behrica**
I looked at the commented out classifier and worked a bit to understand the code and I enabled the first one.

### CODE REVIEW

File: src/tech/v3/libs/smile/protocols.clj [hunk 1]
```diff
@@ -55,14 +55,16 @@
...
     (->> (:options metadata)
-         (reduce (fn [^Properties props {:keys [name default]}]
-                   (.put props (format "%s.%s"
-                                       pname-stem
-                                       (.replace ^String (clojure.core/name name)
-                                                 "-" "."))
-                         (str (dtype/cast (or (get options name)
-                                              (resolve-default default dataset))
-                                          (dtype/get-datatype default))))
+         (reduce (fn [^Properties props {:keys [name default lookup-table]}]
```
Comment (behrica) [inline src/tech/v3/libs/smile/protocols.clj:58]: Made :lookup-table working

File: src/tech/v3/libs/smile/classification.clj [hunk 2]
```diff
@@ -87,23 +88,32 @@
...
+               :type :int32
+               :default 100}
+              {:name :node-size
+               :type :int32
+               :default 1}
+              {:name :max-depth
+               :type :int32 
+               :default 20}
+              {:name :split-rule
+               ;; :type :enumeration
```
Comment (behrica) [inline src/tech/v3/libs/smile/classification.clj:108]: The :type and :class-type keys are not used any where, correct ?

**Comment by behrica** [comment]
I did not add anything for the grid-search,
as I was not sure if it is correct to specify the grid-search intervals on the level of the model description.

Is this not something, which gets defined on a case-by-case base, or is there a concept  of "default gridsearch intervals" of a model ?

File: src/tech/v3/libs/smile/classification.clj (hunk 2 above)
Comment (cnuernber) [reply src/tech/v3/libs/smile/classification.clj:108]: correct, not any more.

**Comment by cnuernber** [comment]
There is a concept of default gridsearch intervals; the default parameters are used.  The smile code could be refactored and gridsearch-options renamed to hyperparameters as that matches the public ml api.

In any case, for a given model the hyperparameters registered are used to construct the gridsearch.

**Comment by cnuernber** [comment]
Are you satisfied with this version?  this PR looks good to me.

**Comment by behrica** [comment]
Maybe I should add some default grid search parameters.
Where did you get the from in the past ?

**Comment by cnuernber** [comment]
I would read the smile source code and see which parameters seemed like the most important to search over.  These are generally constructor arguments in Smile and now Smile takes a generic property map.

**Comment by behrica** [comment]
Yes, I found the parameters:

```
 int maxDepth = Integer.parseInt(prop.getProperty("smile.cart.max.depth", "20"));
        int maxNodes = Integer.parseInt(prop.getProperty("smile.cart.max.nodes", String.valueOf(data.size() / 5)));
        int nodeSize = Integer.parseInt(prop.getProperty("smile.cart.node.size", "5"));
```

but what's about its range ?

**Comment by cnuernber** [comment]
Good question/  We really need a range of problems to know the answer to that one.

**Comment by behrica** [comment]
I added "some" intervals.
They can be overridden in any case, so they are a quick start.

**Comment by behrica** [comment]
If you don't have further comments, I am done with it.

**Comment by cnuernber** [comment]
Thanks, this looks fine to me also.  
It would be cool to have LDA working but I think that is one we would have to do from scratch ourselves.

**Pull request merged**